source/extensions/filters/network/thrift_proxy/router/router_ratelimit_impl.cc

```diff
@@ -0,0 +1,162 @@
+#include "extensions/filters/network/thrift_proxy/router/router_ratelimit_impl.h"
```

Comment:
just a thought: this logic is extremely close to that in the http router, with the exception of headermap v mesagemetadata. Couldn't we template that out and reuse the logic? Or use a better fit c++ pattern to reuse the code in both places? wdyt?

Reply:
Probably. It turns out the HTTP router and Thrift router have a lot in common and there could probably be a common base class, but it seemed to large a project to take on so I opted not to try. I expect some day there might be a third router and maybe it'll make sense then.